Args in experiment:
Namespace(activation='gelu', anomaly_ratio=0.25, batch_size=32, c_out=822, checkpoints='./checkpoints/', d_ff=2048, d_layers=1, d_model=512, data='custom', data_path='pems04_d.csv', dec_in=822, des='Exp', devices='0,1,2,3', distil=True, dropout=0, e_layers=2, embed='timeF', enc_in=822, factor=3, features='M', freq='h', gap_day=365, gpu=0, inverse=False, is_training=1, itr=1, label_len=48, learning_rate=0.0005, loss='MSE', lradj='type3', mask_rate=0.25, model='DLinear', model_id='pems04_d_96_720', moving_avg=25, n_heads=8, num_kernels=6, num_workers=10, output_attention=False, p_hidden_dims=[128, 128], p_hidden_layers=2, patience=5, pred_len=720, root_path='../../data/pems/', samle_rate=1.0, sample_seed=7, seasonal_patterns='Monthly', seq_len=96, target='', task_name='long_term_forecast', top_k=5, train_epochs=200, train_seed=2024, use_amp=False, use_gpu=True, use_multi_gpu=False)
Use GPU: cuda:0
>>>>>>>start training : long_term_forecast_pems04_d_96_720_DLinear_custom_ftM_sl96_ll48_pl720_dm512_nh8_el2_dl1_df2048_fc3_ebtimeF_dtTrue_srate1.0_sseed7_trainseed2024_Exp_0>>>>>>>>>>>>>>>>>>>>>>>>>>
Traceback (most recent call last):
  File "run.py", line 158, in <module>
    exp.train(setting)
  File "/g/data/hn98/du/exlts/hourdayweek/exp/exp_long_term_forecasting.py", line 84, in train
    train_data, train_loader = self._get_data(flag='train')
  File "/g/data/hn98/du/exlts/hourdayweek/exp/exp_long_term_forecasting.py", line 31, in _get_data
    data_set, data_loader = data_provider(self.args, flag)
  File "/g/data/hn98/du/exlts/hourdayweek/data_provider/data_factory.py", line 74, in data_provider
    data_set = Data(
  File "/g/data/hn98/du/exlts/hourdayweek/data_provider/data_loader.py", line 320, in __init__
    self.__read_data__()
  File "/g/data/hn98/du/exlts/hourdayweek/data_provider/data_loader.py", line 324, in __read_data__
    df_raw = pd.read_csv(os.path.join(self.root_path,
  File "/jobfs/116408127.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/jobfs/116408127.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/util/_decorators.py", line 331, in wrapper
    return func(*args, **kwargs)
  File "/jobfs/116408127.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 950, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/jobfs/116408127.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 605, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/jobfs/116408127.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1442, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/jobfs/116408127.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1735, in _make_engine
    self.handles = get_handle(
  File "/jobfs/116408127.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/common.py", line 856, in get_handle
    handle = open(
FileNotFoundError: [Errno 2] No such file or directory: '../../data/pems/pems04_d.csv'
Args in experiment:
Namespace(activation='gelu', anomaly_ratio=0.25, batch_size=32, c_out=822, checkpoints='./checkpoints/', d_ff=2048, d_layers=1, d_model=512, data='custom', data_path='pems04_d.csv', dec_in=822, des='Exp', devices='0,1,2,3', distil=True, dropout=0, e_layers=2, embed='timeF', enc_in=822, factor=3, features='M', freq='h', gap_day=365, gpu=0, inverse=False, is_training=1, itr=1, label_len=48, learning_rate=0.0005, loss='MSE', lradj='type3', mask_rate=0.25, model='DLinear', model_id='pems04_d_96_720', moving_avg=25, n_heads=8, num_kernels=6, num_workers=10, output_attention=False, p_hidden_dims=[128, 128], p_hidden_layers=2, patience=5, pred_len=720, root_path='../../data/pems/', samle_rate=1.0, sample_seed=7, seasonal_patterns='Monthly', seq_len=96, target='', task_name='long_term_forecast', top_k=5, train_epochs=200, train_seed=2024, use_amp=False, use_gpu=True, use_multi_gpu=False)
Use GPU: cuda:0
>>>>>>>start training : long_term_forecast_pems04_d_96_720_DLinear_custom_ftM_sl96_ll48_pl720_dm512_nh8_el2_dl1_df2048_fc3_ebtimeF_dtTrue_srate1.0_sseed7_trainseed2024_Exp_0>>>>>>>>>>>>>>>>>>>>>>>>>>
Traceback (most recent call last):
  File "run.py", line 158, in <module>
    exp.train(setting)
  File "/g/data/hn98/du/exlts/hourdayweek/exp/exp_long_term_forecasting.py", line 84, in train
    train_data, train_loader = self._get_data(flag='train')
  File "/g/data/hn98/du/exlts/hourdayweek/exp/exp_long_term_forecasting.py", line 31, in _get_data
    data_set, data_loader = data_provider(self.args, flag)
  File "/g/data/hn98/du/exlts/hourdayweek/data_provider/data_factory.py", line 74, in data_provider
    data_set = Data(
  File "/g/data/hn98/du/exlts/hourdayweek/data_provider/data_loader.py", line 320, in __init__
    self.__read_data__()
  File "/g/data/hn98/du/exlts/hourdayweek/data_provider/data_loader.py", line 324, in __read_data__
    df_raw = pd.read_csv(os.path.join(self.root_path,
  File "/jobfs/116443174.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/jobfs/116443174.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/util/_decorators.py", line 331, in wrapper
    return func(*args, **kwargs)
  File "/jobfs/116443174.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 950, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/jobfs/116443174.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 605, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/jobfs/116443174.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1442, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/jobfs/116443174.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/parsers/readers.py", line 1735, in _make_engine
    self.handles = get_handle(
  File "/jobfs/116443174.gadi-pbs/timesnet/lib/python3.8/site-packages/pandas/io/common.py", line 856, in get_handle
    handle = open(
FileNotFoundError: [Errno 2] No such file or directory: '../../data/pems/pems04_d.csv'
Args in experiment:
Namespace(activation='gelu', anomaly_ratio=0.25, batch_size=32, c_out=822, checkpoints='./checkpoints/', d_ff=2048, d_layers=1, d_model=512, data='custom', data_path='pems04_d.csv', dec_in=822, des='Exp', devices='0,1,2,3', distil=True, dropout=0, e_layers=2, embed='timeF', enc_in=822, factor=3, features='M', freq='h', gap_day=365, gpu=0, inverse=False, is_training=1, itr=1, label_len=48, learning_rate=0.0005, loss='MSE', lradj='type3', mask_rate=0.25, model='DLinear', model_id='pems04_d_96_720', moving_avg=25, n_heads=8, num_kernels=6, num_workers=10, output_attention=False, p_hidden_dims=[128, 128], p_hidden_layers=2, patience=5, pred_len=720, root_path='../../data/pems/', samle_rate=1.0, sample_seed=7, seasonal_patterns='Monthly', seq_len=96, target='', task_name='long_term_forecast', top_k=5, train_epochs=200, train_seed=2024, use_amp=False, use_gpu=True, use_multi_gpu=False)
Use GPU: cuda:0
>>>>>>>start training : long_term_forecast_pems04_d_96_720_DLinear_custom_ftM_sl96_ll48_pl720_dm512_nh8_el2_dl1_df2048_fc3_ebtimeF_dtTrue_srate1.0_sseed7_trainseed2024_Exp_0>>>>>>>>>>>>>>>>>>>>>>>>>>
train 4759
val 79
test 873
	iters: 100, epoch: 1 | loss: 0.9933694
	speed: 0.1083s/iter; left time: 3194.1740s
Epoch: 1 cost time: 15.847771167755127
Epoch: 1, Steps: 148 | Train Loss: 0.8784954 Vali Loss: 0.5243105 Test Loss: 0.8467043
Validation loss decreased (inf --> 0.524311).  Saving model ...
	iters: 100, epoch: 2 | loss: 0.8683395
	speed: 0.2330s/iter; left time: 6839.6850s
Epoch: 2 cost time: 15.699995756149292
Epoch: 2, Steps: 148 | Train Loss: 0.8532930 Vali Loss: 0.5237972 Test Loss: 0.8370073
Validation loss decreased (0.524311 --> 0.523797).  Saving model ...
Traceback (most recent call last):
  File "/jobfs/116444158.gadi-pbs/timesnet/lib/python3.8/site-packages/torch/serialization.py", line 373, in save
    return
  File "/jobfs/116444158.gadi-pbs/timesnet/lib/python3.8/site-packages/torch/serialization.py", line 260, in __exit__
    self.buffer.flush()
OSError: [Errno 122] Disk quota exceeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "run.py", line 158, in <module>
    exp.train(setting)
  File "/g/data/hn98/du/exlts/hourdayweek/exp/exp_long_term_forecasting.py", line 214, in train
    early_stopping(vali_loss, self.model, path)
  File "/g/data/hn98/du/exlts/hourdayweek/utils/tools.py", line 51, in __call__
    self.save_checkpoint(val_loss, model, path)
  File "/g/data/hn98/du/exlts/hourdayweek/utils/tools.py", line 57, in save_checkpoint
    torch.save(model.state_dict(), path + '/' + 'checkpoint.pth')
  File "/jobfs/116444158.gadi-pbs/timesnet/lib/python3.8/site-packages/torch/serialization.py", line 374, in save
    _legacy_save(obj, opened_file, pickle_module, pickle_protocol)
  File "/jobfs/116444158.gadi-pbs/timesnet/lib/python3.8/site-packages/torch/serialization.py", line 214, in __exit__
    self.file_like.close()
OSError: [Errno 122] Disk quota exceeded
Args in experiment:
Namespace(activation='gelu', anomaly_ratio=0.25, batch_size=32, c_out=822, checkpoints='./checkpoints/', d_ff=2048, d_layers=1, d_model=512, data='custom', data_path='pems04_d.csv', dec_in=822, des='Exp', devices='0,1,2,3', distil=True, dropout=0, e_layers=2, embed='timeF', enc_in=822, factor=3, features='M', freq='h', gap_day=365, gpu=0, inverse=False, is_training=1, itr=1, label_len=48, learning_rate=0.0005, loss='MSE', lradj='type3', mask_rate=0.25, model='DLinear', model_id='pems04_d_96_720', moving_avg=25, n_heads=8, num_kernels=6, num_workers=10, output_attention=False, p_hidden_dims=[128, 128], p_hidden_layers=2, patience=5, pred_len=720, root_path='../../data/pems/', samle_rate=1.0, sample_seed=7, seasonal_patterns='Monthly', seq_len=96, target='', task_name='long_term_forecast', top_k=5, train_epochs=200, train_seed=2024, use_amp=False, use_gpu=True, use_multi_gpu=False)
Use GPU: cuda:0
>>>>>>>start training : long_term_forecast_pems04_d_96_720_DLinear_custom_ftM_sl96_ll48_pl720_dm512_nh8_el2_dl1_df2048_fc3_ebtimeF_dtTrue_srate1.0_sseed7_trainseed2024_Exp_0>>>>>>>>>>>>>>>>>>>>>>>>>>
train 4759
val 79
test 873
	iters: 100, epoch: 1 | loss: 0.9933694
	speed: 0.1095s/iter; left time: 3230.1069s
Epoch: 1 cost time: 16.01408886909485
Epoch: 1, Steps: 148 | Train Loss: 0.8784954 Vali Loss: 0.5243105 Test Loss: 0.8467043
Validation loss decreased (inf --> 0.524311).  Saving model ...
	iters: 100, epoch: 2 | loss: 0.8683395
	speed: 0.2349s/iter; left time: 6896.1360s
Epoch: 2 cost time: 15.711600303649902
Epoch: 2, Steps: 148 | Train Loss: 0.8532930 Vali Loss: 0.5237972 Test Loss: 0.8370073
Validation loss decreased (0.524311 --> 0.523797).  Saving model ...
	iters: 100, epoch: 3 | loss: 0.7374974
	speed: 0.2357s/iter; left time: 6882.2745s
Epoch: 3 cost time: 15.873997926712036
Epoch: 3, Steps: 148 | Train Loss: 0.8498742 Vali Loss: 0.5243461 Test Loss: 0.8334295
EarlyStopping counter: 1 out of 5
	iters: 100, epoch: 4 | loss: 0.8968607
	speed: 0.2348s/iter; left time: 6822.0150s
Epoch: 4 cost time: 15.748441934585571
Epoch: 4, Steps: 148 | Train Loss: 0.8498420 Vali Loss: 0.5243827 Test Loss: 0.8332475
EarlyStopping counter: 2 out of 5
	iters: 100, epoch: 5 | loss: 0.7931261
	speed: 0.2346s/iter; left time: 6780.6238s
Epoch: 5 cost time: 15.782628297805786
Epoch: 5, Steps: 148 | Train Loss: 0.8492890 Vali Loss: 0.5237553 Test Loss: 0.8334963
Validation loss decreased (0.523797 --> 0.523755).  Saving model ...
	iters: 100, epoch: 6 | loss: 0.7518842
	speed: 0.2346s/iter; left time: 6746.5531s
Epoch: 6 cost time: 15.72965407371521
Epoch: 6, Steps: 148 | Train Loss: 0.8494022 Vali Loss: 0.5233092 Test Loss: 0.8326575
Validation loss decreased (0.523755 --> 0.523309).  Saving model ...
	iters: 100, epoch: 7 | loss: 0.7753155
	speed: 0.2362s/iter; left time: 6759.7328s
Epoch: 7 cost time: 15.89558458328247
Epoch: 7, Steps: 148 | Train Loss: 0.8496838 Vali Loss: 0.5237356 Test Loss: 0.8320764
EarlyStopping counter: 1 out of 5
	iters: 100, epoch: 8 | loss: 0.8812037
	speed: 0.2356s/iter; left time: 6706.6924s
Epoch: 8 cost time: 15.782488584518433
Epoch: 8, Steps: 148 | Train Loss: 0.8499564 Vali Loss: 0.5250793 Test Loss: 0.8355641
EarlyStopping counter: 2 out of 5
	iters: 100, epoch: 9 | loss: 0.8282664
	speed: 0.2348s/iter; left time: 6647.4717s
Epoch: 9 cost time: 15.69906210899353
Epoch: 9, Steps: 148 | Train Loss: 0.8494158 Vali Loss: 0.5233177 Test Loss: 0.8351181
EarlyStopping counter: 3 out of 5
	iters: 100, epoch: 10 | loss: 0.8581402
	speed: 0.2345s/iter; left time: 6604.3347s
Epoch: 10 cost time: 15.735911846160889
Epoch: 10, Steps: 148 | Train Loss: 0.8495402 Vali Loss: 0.5235690 Test Loss: 0.8332332
EarlyStopping counter: 4 out of 5
	iters: 100, epoch: 11 | loss: 0.8582134
	speed: 0.2352s/iter; left time: 6590.8623s
Epoch: 11 cost time: 15.742654085159302
Epoch: 11, Steps: 148 | Train Loss: 0.8493137 Vali Loss: 0.5233930 Test Loss: 0.8349891
EarlyStopping counter: 5 out of 5
Early stopping
>>>>>>>testing : long_term_forecast_pems04_d_96_720_DLinear_custom_ftM_sl96_ll48_pl720_dm512_nh8_el2_dl1_df2048_fc3_ebtimeF_dtTrue_srate1.0_sseed7_trainseed2024_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 873
test shape: (873, 1, 720, 822) (873, 1, 720, 822)
test shape: (873, 720, 822) (873, 720, 822)
mse:0.8326550126075745, mae:0.6245192289352417
>>>>>>>Overall time: 287 seconds<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
